### PR TITLE
Add static host user tctl

### PR DESCRIPTION
**Pull request by atburke**
This change adds `tctl` support for the static host user resource.

Part of #42712.

Changelog: Added static host user resource

### CODE REVIEW

**Review by rosstimothy — APPROVED** [review]

File: tool/tctl/common/collection.go [hunk 1]
```diff
@@ -1733,3 +1735,60 @@ func (c *spiffeFederationCollection) writeText(w io.Writer, verbose bool) error
...
+				gid,
+			})
+		}
+	}
+	headers := []string{"Login", "Node Labels", "Node Expression", "Groups", "Uid", "Gid"}
+	var t asciitable.Table
+	if verbose {
+		t = asciitable.MakeTable(headers, rows...)
+	} else {
+		t = asciitable.MakeTableWithTruncatedColumn(headers, rows, "Node Expression")
```
Comment (GavinFrazar) [inline tool/tctl/common/collection.go:1783]: Why not truncate the "Labels" column like other resource types do?

File: tool/tctl/common/collection.go [hunk 2]
```diff
@@ -1733,3 +1735,60 @@ func (c *spiffeFederationCollection) writeText(w io.Writer, verbose bool) error
...
+			}
+			rows = append(rows, []string{
+				item.GetMetadata().Name,
+				printSortedStringSlice(labelStrings),
+				matcher.NodeLabelsExpression,
+				printSortedStringSlice(matcher.Groups),
+				uid,
+				gid,
+			})
+		}
```
Comment (GavinFrazar) [inline tool/tctl/common/collection.go:1776]: could I suggest this patch to make use of the common label formatting:

```diff
diff --git a/tool/tctl/common/collection.go b/tool/tctl/common/collection.go
index 9b85969a8b..f8e18d2444 100644
--- a/tool/tctl/common/collection.go
+++ b/tool/tctl/common/collection.go
@@ -43,6 +43,7 @@ import (
 	"github.com/gravitational/teleport/api/types/accesslist"
 	"github.com/gravitational/teleport/api/types/discoveryconfig"
 	"github.com/gravitational/teleport/api/types/externalauditstorage"
+	"github.com/gravitational/teleport/api/types/label"
 	"github.com/gravitational/teleport/api/types/secreports"
 	apiutils "github.com/gravitational/teleport/api/utils"
 	"github.com/gravitational/teleport/lib/asciitable"
@@ -1753,9 +1754,10 @@ func (c *staticHostUserCollection) writeText(w io.Writer, verbose bool) error {
 	for _, item := range c.items {
 
 		for _, matcher := range item.Spec.Matchers {
-			labelStrings := make([]string, 0, len(matcher.NodeLabels))
-			for _, label := range matcher.NodeLabels {
-				labelStrings = append(labelStrings, fmt.Sprintf("%s=[%s]", label.Name, printSortedStringSlice(label.Values)))
+			labelMap := label.ToMap(matcher.NodeLabels)
+			labelStringMap := make(map[string]string)
+			for k, vals := range labelMap {
+				labelStringMap[k] = fmt.Sprintf("[%s]", printSortedStringSlice(vals))
 			}
 			var uid string
 			if matcher.Uid != 0 {
@@ -1767,7 +1769,7 @@ func (c *staticHostUserCollection) writeText(w io.Writer, verbose bool) error {
 			}
 			rows = append(rows, []string{
 				item.GetMetadata().Name,
-				printSortedStringSlice(labelStrings),
+				common.FormatLabels(labelStringMap, verbose),
 				matcher.NodeLabelsExpression,
 				printSortedStringSlice(matcher.Groups),
 				uid,

```

I would make a github suggestion in-line but the diff it renders is very unhelpful to see what changes I'm suggesting

**Review by GavinFrazar — COMMENTED** [review]
mostly LGTM but I left a few comments

File: tool/tctl/common/collection.go (hunk 1 above)
Comment (atburke) [reply tool/tctl/common/collection.go:1785]: My hunch is that the typical node expression will be longer than the typical labels.

**Review by GavinFrazar — APPROVED** [review]

**Pull request merged**